Lib/importlib/_bootstrap_external.py

```diff
@@ -908,7 +908,7 @@ class ExtensionFileLoader(FileLoader, _LoaderBasics):
 
     def __init__(self, name, path):
         self.name = name
-        self.path = path
+        self.path = path.__fspath__() if hasattr(path, '__fspath__') else path
```

Comment:
It's probably better to call `super().__init__(name, path)` than to duplicate this just for extension modules.